packages/office-addin-cli/src/parse_json_script_variable.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+const readJson = require('read-package-json-fast');
+
+let jsonPackage : any = undefined;
+
+export async function parseJSONScriptVariable(variable: string): Promise <string | undefined> {

```

Comment:
Consider the name of this function. If you were reading it, would you know what json is being parsed. Is parsing even that important here? When I consider this, I try to say to myself: what does this function do? It returns the value of the script from **package.json**. So a better name would be `getScriptFromPackageJson()` or `getPackageJsonScript()`. The other thing is it does not specify the package.json file, but this is implicit because of the NPM definition of `process.env.npm_package_json`. Do we want it to be implicit? There is no mention of NPM definition of environment variables. Should there be?

Perhaps it would be better to characterize this function as reading an NPM environment variable, e.g. `getNpmEnvVariable(name: string)` which takes "npm_package_script_*" and knows to look it up. This most closely replaces the direct behavior of `process.env.npm_package_script_dev_server` with `getNpmEnvVariable("npm_package_script_dev_server")` which is a simple replacement.

In terms of the implementation, if the env var name is passed in, the first thing it should do is see if that is defined and return it if so. 

In the next part, if it has not already done so, it needs to read and parse the package.json file. If we really only need to support reading the scripts, I would go ahead and make a helper function which takes the file path, parses the scripts, and generates the script env vars, so you can just keep an object with the script environment variables. It seems that rather than hold onto the contents of package.json or the script values, it could instead just write the environment variables. If it does this, then later checks would just find them defined and do nothing. That has the advantage of not really holding anything extra.

Reply:
I just did those changes. Changed the function name to a more meaningful name and also the code just storing what is needed.